torch/_inductor/codegen/triton.py

```diff
@@ -1096,14 +1102,25 @@ def load(self, name: str, index: sympy.Expr):
         original_index = index
         index, mask_vars, mask, expand_str = self.indexing(index)
 
-        if "rmask" in mask and not self.persistent_reduction:
-            # This eviction policy heuristic is untested.
-            # ptillet suggested we should try only doing this for
-            # the first N-1 loops and not for the final loop.
-            ep = ", eviction_policy='evict_last'"
+
+        # Keep the variable in cache if we are going to reuse it
+        # TODO(lezcano) We could potentially do better
+        # https://github.com/pytorch/pytorch/pull/91316#issuecomment-1364680622
+        # Keep the variable in cache last if all the folowing hold
+        #  1) We are in a reduction loop
+        #  2) Its not its last use
+        #  3) This load will not be lifted to body the body
+        #  4) We are not broadcasting
+        if self.inside_reduction and not self.persistent_reduction:
+            names = self.mutations or {name}
```

Comment:
Won't `self.mutations` refer to every buffer used in the current loop, not just the one being loaded here?